### ISSUE TITLE
Interactive clipping plugin

### ISSUE THREAD

**Issue by HristijanJohn** [xBimTeam/XbimWebUI#310]
Hello,
I am using the newest version of the xbim viewer package. Is there any example how can I use interactive clipping plugin. In the previous versions it was very simple with calling just viewer.clip() function. But with the newest version I am getting the following error "Cutting plane not well defined". I checked docs for the interactive clipping plugin, but  it didn't help me a lot. I also saw examples where we can use point and normal vector as parameters, but I need the one where the user can mark the cutting planes with mouse moving. Any help will be appreciated. Thanks.

**Comment by andyward** [comment]
Hi @HristijanJohn sorry nobody got back. Did you get it resolved?

For reference there are 3 different variations of the clipping API:

1. A single plane 

https://docs.xbim.net/XbimWebUI/classes/_viewer_.viewer.html#clip

`viewer.clip(point: xyz, vector: xyz);`

This applies a single clip plane at the point in the direction of the vector.

e.g. `viewer.clip([-8,1,0], [-1,0,0]) // Clip on x plane at -8, 1`

![image](https://user-images.githubusercontent.com/923699/106302062-5ea96e80-6250-11eb-9c69-c849e91ffa22.png)

2. A pair of planes

https://docs.xbim.net/XbimWebUI/classes/_viewer_.viewer.html#setclippingplanea

`viewer.setClippingPlaneA(plane: xyzOffset);`
and its pair setClippingPlaneB(plane)

e.g. section a building floor
```javascript
viewer.setClippingPlaneA([0,0,-1, 3]); // Clip Above 3 (m)
viewer.setClippingPlaneB([0,0,1, 0]) // Clip Below 0
```

![image](https://user-images.githubusercontent.com/923699/106301662-e2169000-624f-11eb-8643-2684cddc7c62.png)

3. Apply a sectionbox (cube)

https://docs.xbim.net/XbimWebUI/classes/_viewer_.viewer.html#sectionbox

This lets you build a box around a centerpoint - the equivalent of 6 x planes. 

```javascript
viewer.sectionBox.location=[-10,-12,0]
viewer.lengthX = 10;
viewer.lengthY = 10;
viewer.lengthZ = 10;
viewer.rotateZ = 35;
```

![image](https://user-images.githubusercontent.com/923699/106308369-83094900-6258-11eb-8db5-f241d004d9e4.png)

In terms of interactive clipping there is a sample plugin at https://github.com/xBimTeam/XbimWebUI/tree/master/src/plugins/InteractiveClipping

... but it will need updating to work with the 2 newer clipping strategies.

**Comment by HristijanJohn** [comment]
Hello I finally succeed to implement interactive clipping plugin into my project. The general problem was the normal vector that I am getting as a result from vector cross operation. The result coordinates were huge numbers. So I used vec3.normalize function to get single vector with the same direction and it's working good now. The changed code is below:
 var BA = vec3.subtract(vec3.create(), A, B);
 var BC = vec3.subtract(vec3.create(), C, B);
var N = vec3.cross(vec3.create(), BA, BC);

viewer.unclip();
viewer.clip(B, vec3.normalize(vec3.create(),N));
